dev/AppLifecycle/AppInstance.cpp

```diff
@@ -33,22 +34,32 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
         // Push past the '----' commandline argument prefix.
         argsStart += 4;
 
-        // We explicitly use find_first_of here, so that the resulting data may contain : as a valid character.
-        auto argsEnd = commandLine.find_first_of(L":", argsStart);
+        auto argsEnd = commandLine.find_first_of(L" ", argsStart);
```

Comment:
Command line parsing is complicated (more so than one might think). Use [CommandLineToArgvW](https://docs.microsoft.com/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw) to do the heavy lifting.

Reply:
It depends. If the command line is

```
----a:b"c\"d"e
```

is the expected parse
1. contract = `a`, data = `b"c\"d"e`; or
2. contract = `a`, data = `bc"de`
CommandLineToArgvW will give you version 2.

Also, it seems that a space is not required before the first dash, so we will treat

```
contoso.exe hello----dolly.txt
```

as
* contract = `dolly.txt`, data = empty

which could be surprising if somebody has a file named "hello----dolly.txt".